### ISSUE TITLE
casting a two column data.table

### ISSUE THREAD

**Issue by smartinsightsfromdata** [Rdatatable/data.table#1070]
This is my reproducible example.  I have the following data.table:
```R
dt <- data.table(month.abb[1:5], val=0)
```
I want to have the months in the columns (a special case of transpose if you will). I use dcast and I get an error.
```
dcast.data.table(dt,... ~ label, value.var="activityunits", sum)
 Error in dcast.data.table(dt, ... ~ label, value.var = "activityunits",  : 
   LHS of formula evaluates to 'character(0)', invalid formula.
```
If I try with a data.frame I get the result I need:
```
df <- as.data.frame(dt)
dcast(df, ... ~ label, value.var="activityunits", sum)
  . Apr Feb Jan Mar May
1 .   0   0   0   0   0
```

(of course this assumes that data.table has implemented ... which perhaps is not the case?)

**Comment by arunsrinivasan** [comment]
Please read [this](https://github.com/Rdatatable/data.table/blob/master/Contributing.md).

Your example data doesn't contain neither `label` nor `activityunits`. Could you please fix that?
Also, are you suing `reshape2` or `reshape`? And which version of `data.table` are you on?

**Comment by smartinsightsfromdata** [comment]
Apologies!
This is using ```data.table v.1.9.4``` - ```reshape2  v.1.4.1```
Again the example:
```R
>dt <- data.table(label= month.abb[1:5], val=0)
>dt
   label val
1:   Jan   0
2:   Feb   0
3:   Mar   0
4:   Apr   0
5:   May   0
>dcast.data.table(dt,... ~ label, value.var="val", sum)
Error in dcast.data.table(dt, ... ~ label, value.var = "val", sum) : 
  LHS of formula evaluates to 'character(0)', invalid formula.

>df <- as.data.frame(dt)
>df <- reshape2::dcast(dt,... ~ label, value.var="val", sum)
>df
  . Apr Feb Jan Mar May
1 .   0   0   0   0   0
```

**Comment by arunsrinivasan** [comment]
Also fixed in commit 25a74dfd8.

```R
dcast.data.table(dt,... ~ label, value.var="val", sum)
#    . Apr Feb Jan Mar May
# 1: .   0   0   0   0   0
```

PS: no need to use `dcast.data.table` anymore in `v1.9.5+`.
